config/foreman.hiera/tuning/sizes/medium.yaml

```diff
@@ -1,4 +1,8 @@
 ---
+apache::mod::event::serverlimit: 64
+apache::mod::event::maxrequestworkers: 1024
+apache::mod::event::maxrequestsperchild: 4000
```

Comment:
This seems quite a jump from default to medium, what is your thinking there?

Reply:
If the question is about `ServerLimit` and `MaxRequestWorkers`, this tuning profile already has `apache::mod::prefork::serverlimit: 1024` which allows up to 1024 single threaded httpd worker processes.

For event_mpm case, `ServerLimit * ThreadsPerChild = MaxRequestWorkers` and we have `ThreadsPerChild = 16`, so matching the # of threads we have already for mpm_prefork implies `ServerLimit * 16 = 1024`, so `ServerLimit =64`

If the question is about `MaxRequestsPerChild`, the default is 0 which disables worker recycling entirely. We previously added `apache::mod::prefork::maxrequestsperchild: 4000` to protect against memory leaks, so matching this value with the event mpm makes sense because threads are able to share memory within each worker process and this is a _per worker_ setting.

I think it's a likely oversight that `MaxRequestsPerChild` was previously not set in the default profile -- if we are concerned about the possibility of memory leaks then we should protect against them in the default profile as well. If you agree, I'll push an update adding that to the default as well.

I was also surprised to find that the same limits are already used in medium through extra-extra-large deployment sizes, and was tempted to actually scale them up and down with the profile sizes, but opted instead to heed the previous advice about changing too much at one time.

Reply:
Appreciate the detailed explanation, that all makes sense. I wouldn't worry about updating prefork configuration - we made it this far and EL7 support is set to be removed really soon.

Reply:
FWIW, now that we don't run `mod_passenger` and `mod_wsgi` inside Apache, I'm less concerned about memory leaks and we could see whether increasing `MaxRequestsPerChild` or even setting it to `0` would make sense.

But obviously, this doesn't belong into *THIS* PR ;-)